From 03d20260f3dc08537111fed48a5672734af6d848 Mon Sep 17 00:00:00 2001
From: Joyce McIntosh <joyce.mcintosh@nexenta.com>
Date: Thu, 7 Feb 2019 16:40:19 -0700
Subject: [PATCH] NEX-15740 NFS deadlock in rfs4_compound with hundreds of
 threads waiting for lock owned by rfs4_op_rename (lint fix)

---
 usr/src/uts/common/fs/nfs/nfs4_srv.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/usr/src/uts/common/fs/nfs/nfs4_srv.c b/usr/src/uts/common/fs/nfs/nfs4_srv.c
index b70524fff7..8b27a4be2a 100644
--- a/usr/src/uts/common/fs/nfs/nfs4_srv.c
+++ b/usr/src/uts/common/fs/nfs/nfs4_srv.c
@@ -4595,7 +4595,8 @@ rfs4_op_rename(nfs_argop4 *argop, nfs_resop4 *resop, struct svc_req *req,
 				if ((srcvp == targvp) || (tvp != targvp)) {
 					cmn_err(CE_WARN, "rfs4_op_rename: "
 					    "srcvp %p, targvp: %p, tvp: %p",
-					    srcvp, targvp, tvp);
+					    (void *)srcvp, (void *)targvp,
+					    (void *)tvp);
 				}
 			} else {
 				VN_RELE(tvp);
@@ -4621,7 +4622,7 @@ rfs4_op_rename(nfs_argop4 *argop, nfs_resop4 *resop, struct svc_req *req,
 		/* DEBUG data */
 		if (RW_READ_HELD(&tvp->v_nbllock)) {
 			cmn_err(CE_WARN, "rfs4_op_rename: "
-			    "RW_READ_HELD(%p)", tvp);
+			    "RW_READ_HELD(%p)", (void *)tvp);
 		}
 
 		/* The file is gone and so should the state */
-- 
2.17.2 (Apple Git-113)

